content/articles/build-tensorflow-from-source/index.md

```diff
@@ -0,0 +1,178 @@
+
+
+#### Introducing custom features 
+
+When we talk about custom features, it ranges from the slightest modification to a larger sub-system taking care of additional controls. As discussed earlier, we are bridging the gap between coding the logic and making them OS specific distributions. OS may not be the only parameter that changes. There are several parameters such as hardware architecture, instruction set architecture (ISA), etc.
+
+#### Lack of availability of pre-built executables
+
+Let's say you have built an OS of your own. It is an extension of Kali Linux, and you have introduced several commands and changed the core algorithms under the hood of the OS. The scheduling routines implemented are optimized for repetitive tasks. In such a scenario, the ability to generate executables is of utmost importance. 
+
+
+### Building TensorFlow 2.5.0 from Source using Bazelisk
+
+The task of building TensorFlow 2.5.0 from source in Windows is a tedious one. The number of variables that need to sit right to build successfully is significantly high. Therefore, we will proceed with the Windows Subsystem for Linux (WSL). To install WSL in Windows, follow the following [tutorial](https://www.windowscentral.com/install-windows-subsystem-linux-windows-10).
+
+When given a prompt to download Ubuntu WSL from the Windows Store, choose Ubuntu 18.04 as the OS release.
+
+On the other hand, if you are using the Ubuntu system, then you can skip the above steps:
+
+Let us initially install `curl`, which will be used to download `bazelisk.`
+
+```bash
+$ sudo apt update
+$ sudo apt full-upgrade
+$ sudo apt install curl
+```
+
+#### Bazelisk
+
+Unlike Linux distributions, Google decided to go ahead with their software build system called Bazel. Bazel is a fast, scalable, multi-language, and extensible build system as defined on the official website. It is used in place of commands `cmake`.
+
+Bazelisk is a wrapper written in Go that automatically downloads the required version of Bazel. Unfortunately, installing Bazel manually is a time-consuming and error-prone path. After many errors, I finally landed at Bazelisk.
+
+```bash
+# Get the Bazelisk file from https://github.com/bazelbuild/bazelisk/releases/ 
+$ sudo curl -Lo /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.10.0/bazelisk-linux-amd64
+$ sudo chmod +x /usr/local/bin/bazel
+```
+Once Bazel is installed using Bazelisk, we can use the following command to check the version of Bazel installed. 
+
+```bash
+# This should work and print a Bazelisk and Bazel version.
+$ bazel version
+
+Bazelisk version: v1.10.0
+Build label: 1.10.0
+```
+Once we have reached this stage, we move to the official guide on the TensorFlow documentation. In the docs, there is a list of compatible versions of Bazel and TensorFlow versions. 
+
+![](/engineering-education/build-tensorflow-from-source/tensorflow-bazel-compatibility.png)
+
```

Comment:
Should this part [....] be empty?

Reply:
Sorry. My bad. I had left it to be completed in the end. Will update it.

Reply:
Can you review the article, so that I make all the necessary changes altogether at once?

Reply:
It is a well-written article. I've already reviewed. That's the only part I was sure about. You can edit that bit then we can proceed for final review.

Reply:
@Bayler I have made the requested edits.

Reply:
Noted. Let me finalize the review